### PR TITLE
Update SSO guide with OIDC improvements

### PR DESCRIPTION
**Pull request by IanHakes-okta**
## Description:
- **What's changed?** 
Update the sign on guide with improved guidance around using OIDC for OIN applications.

Also fixes "he" usage in the SAML concept topic.

- **Is this PR related to a Monolith release?** 
No
### Resolves:

* [OKTA-299406](https://oktainc.atlassian.net/browse/OKTA-299406)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 ### Prepare an OIDC integration
 
+Okta uses a multi-tenant local credential system for OIDC. Each instance of your app integration inside a customer org has a separate set of OIDC client credentials that are used to access your application.
+
+For example, consider a scenario where your app integration is added to ten separate customer orgs. Seven of those customers create a single instance of your app integration. However, the other three customers each create two separate instances of your app integration so they can use different configuration options. This creates a total of 13 (7 + (3*2)) sets of client credentials for your application that you need to track.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:5]: ten -> 10

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md [hunk 2]
```diff
@@ -9,9 +15,10 @@ Before you create a new OIDC integration in Okta:
    * A single page app (SPA) is a web application that is contained on a single web page. All code is retrieved when the page is loaded initially - the page doesn't reload or refresh. A SPA application cannot keep running on a server.
 -->
 
+1. Have your application developed and tested - with a front-end (for example, JavaScript and HTML) and back-end (for example, middleware and database software) stack, with services available through APIs and accepting HTTP connections.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:18]: Have your application developed and tested with a front-end (for example, JavaScript and HTML) and back-end (for example, middleware and database software) stack, with services available through APIs, and accepting HTTP connections.

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md [hunk 3]
```diff
@@ -9,9 +15,10 @@ Before you create a new OIDC integration in Okta:
...
+1. Based on the [type of application you have built](/docs/concepts/auth-overview/#what-kind-of-client-are-you-building), determine the correct [OAuth 2.0 flow](/docs/concepts/auth-overview/#recommended-flow-by-application-type) that is required below the OIDC identity layer. For OIN app integrations, the OAuth 2.0 flow must be either Web (server side) or a Single Page Application (SPA).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:19]: type of application you have built -> type of application that you have built

for translation

Single Page Application -> Single-Page Application

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/create-your-app/index.md [hunk 4]
```diff
@@ -2,7 +2,9 @@
 title: Create your integration
 ---
 
-After you have your background information, you can use the Okta Admin Console and the Application Integration Wizard to create your SSO integration inside the Okta org associated with your developer account.
+After you have your background information, you can use the Okta Admin Console and the Application Integration Wizard (AIW) to create your SSO integration inside the Okta org associated with your developer account.
+
+>**Note:** Creating your SSO app integration does not automatically make it available in the [OIN](https://www.okta.com/integrations/). After you have created and tested it, you need to [submit your app integration](/docs/guides/submit-app/openidconnect/overview/) to the OIN.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/build-sso-integration/create-your-app/index.md:7]: does not -> doesn't

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/overview/index.md [hunk 5]
```diff
@@ -1,34 +1,42 @@
...
+
+In a typical scenario, your application relies on Okta to act as a multi-tenant Identity Provider (IdP) for your customers' Okta organizations.
+
+An [Okta org](/docs/concepts/okta-organizations/) acts as a container that sets hard boundaries for all users, applications, and other entities associated with a single customer, providing tenant-based isolation.
+
+In developing your SSO app integration, the customer’s Okta org serves as the Authorization Server (OIDC) or as the IdP (SAML).
+
+## Publishing
+
+This guide assumes you intend to develop an app integration and make it public by publishing it in the Okta Integration Network (OIN). If you want to develop a custom app integration that is intended for private deployment within your own company, use the Okta [App Integration Wizard (AIW)](https://help.okta.com/en/prod/okta_help_CSH.htm#ext_Apps_App_Integration_Wizard) to create your app integration.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/build-sso-integration/overview/index.md:40]: This guide assumes that you intend....

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/specify-your-settings/openidconnect/settings.md [hunk 6]
```diff
@@ -2,6 +2,17 @@
...
+
+  The grant types permitted for OIN applications depend on the application type selected:
+
+  * Web:
+    * **Authorization code** &mdash; mandatory for Web application types
+    * Optional. **Refresh token**
+    * Optional. **Implicit (Hybrid)**
+  * SPA:
+    * Implicit &mdash; choose either **Allow ID Token with implicit grant type** or **Allow Access Token with implicit grant type**
+* The **Client acting on behalf of itself** grant type is not permitted in OIN app integrations.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/build-sso-integration/specify-your-settings/openidconnect/settings.md:15]: this would work better as a note - as a bullet, it's kind of confusing.....is it even with "Web"? Is it aligned with "On the General tab"? If so, that's extra confusing - because it's not really a step? The formatting here is hard to follow.

File: packages/@okta/vuepress-site/docs/concepts/saml/index.md [hunk 7]
```diff
@@ -13,7 +13,7 @@ However, with increased collaboration and the move towards the cloud, many appli
...
+Before diving into federated authentication, we need to understand what authentication really means. Authentication defines the way a user is identified and validated through some sort of credentials as part of a sign-in flow. Most applications present a sign-in page to an end user, allowing the user to specify a username and a password. In some cases, additional information may be required to locate the user - like a company ID or a client code. This information allows the application to narrow down the search of the username applicable to the provided info. This is often used to allow the same username to exist across multiple tenants belonging to different customers.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/concepts/saml/index.md:16]: In some cases, additional information may be required to locate the user, like a company ID or a client code.

In our Global English guide - we have information about not using em dashes as a formatting device. I know that we've implemented these in some cases when we discuss parameters in a list, but here in text, we should avoid them. 

"There is nothing inherently wrong with using em dashes in these contexts. However, if we sometimes use em dashes and other times use parentheses to set off the same type of information, the inconsistency interferes with the efficient use of translation memory.”

So basically. It’s not wrong, but it’s also not common. Better to just use a comma and not risk introducing a weird formatting idiosyncracy.

Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/concepts/saml/index.md:16]: Sure thing - I can make this change.
(In my defense, I haven't rewritten this content at all. This was just to remove the use of "him" that someone found.)

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md (hunk 2 above)
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:18]: Have your application developed and tested, with both a front-end (for example, JavaScript and HTML) and a back-end (for example, middleware and databases) software stack, along with services available through APIs, and accepting HTTP connections.

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/specify-your-settings/openidconnect/settings.md (hunk 6 above)
Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-sso-integration/specify-your-settings/openidconnect/settings.md:15]: Good point - I've moved it to a Note

File: packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md [hunk 8]
```diff
@@ -1,17 +1,24 @@
...
 -->
 
+1. Have your application developed and tested, with a front-end (for example, JavaScript and HTML) and back-end (for example, middleware and database software) stack, along with services available through APIs, and accepting HTTP connections.
+1. Based on the [type of application that you have built](/docs/concepts/auth-overview/#what-kind-of-client-are-you-building), determine the correct [OAuth 2.0 flow](/docs/concepts/auth-overview/#recommended-flow-by-application-type) that is required below the OIDC identity layer. For OIN app integrations, the OAuth 2.0 flow must be either Web (server side) or a Single-Page Application (SPA).
 1. Determine the login redirect URIs on your system. A redirect URI is where Okta sends the authentication response and ID token during the sign-in flow. You can specify more than one URI if required.
 1. Your application must support automatic credential rotation. For more information, see the `/keys` section in the [OpenID Connect & OAuth 2.0 API reference](/docs/reference/api/oidc/#key-rotation).
```
Comment (kasturikumthekar-okta) [inline packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:21]: Why is everything 1.

Comment (IanHakes-okta) [reply packages/@okta/vuepress-site/docs/guides/build-sso-integration/before-you-begin/openidconnect/prep.md:21]: That is markdown - shows that they are sequential numbered steps. Using "1" just means that if additional steps are added later, you don't have to renumber everything - markdown figures it out when it processes.
